src/common/datomish/transact.cljc

```diff
@@ -552,12 +567,75 @@
     (:db-after (<? (<with db tx-data)))))
 
 (defn <transact!
+  "Submits a transaction to the database for writing.
+
+  Returns a pair-chan resolving to `[result error]`."
   [conn tx-data]
   {:pre [(conn? conn)]}
-  (let [db (db conn)] ;; TODO: be careful with swapping atoms.
-    (db/in-transaction!
-      db
-      #(go-pair
-         (let [report (<? (<with db tx-data))]
-           (reset! (:current-db conn) (:db-after report))
-           report)))))
+  (let [result (a/chan 1)]
```

Comment:
Oh, later I see you close it after use. Never mind. Think about it if you think it might be useful.